lndclient/lnd_services.go

```diff
@@ -214,25 +263,36 @@ func (s *GrpcLndServices) Close() {
 }
 
 // checkLndCompatibility makes sure the connected lnd instance is running on the
-// correct network.
+// correct network, has the version RPC implemented, is the correct minimal
+// version and supports all required build tags/subservers.
 func checkLndCompatibility(conn *grpc.ClientConn, chainParams *chaincfg.Params,
-	readonlyMac serializedMacaroon, network string) (string, [33]byte,
-	error) {
+	readonlyMac serializedMacaroon, network string,
+	minVersion *verrpc.Version) (string, [33]byte, *verrpc.Version, error) {
 
 	// onErr is a closure that simplifies returning multiple values in the
 	// error case.
-	onErr := func(err error) (string, [33]byte, error) {
+	onErr := func(err error) (string, [33]byte, *verrpc.Version, error) {
 		closeErr := conn.Close()
 		if closeErr != nil {
 			log.Errorf("Error closing lnd connection: %v", closeErr)
 		}
 
-		return "", [33]byte{}, err
+		// Make static error messages a bit less cryptic by adding the
+		// version or build tag that we expect.
+		newErr := fmt.Errorf("lnd compatibility check failed: %v", err)
+		if err == ErrVersionIncompatible || err == ErrBuildTagsMissing {
+			newErr = fmt.Errorf("error checking connected lnd "+
```

Comment:
Now you could just as well return this error from `checkVersionCompatibility` directly? Maybe this just exists for the sake of the unit tests now (nb).

Reply:
Yeah, but I didn't want to rewrite the unit tests and fall back to string matching there, so I left it. This would be a nice use case for the error wrapping, but I'm not sure if we already want to use that broadly?

Reply:
I haven't seen it yet. Let's merge this thing